managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1294,9 +1293,24 @@ public synchronized void updateLedgersIdsComplete(Stat stat) {
             log.debug("[{}] Resending {} pending messages", name, pendingAddEntries.size());
         }
 
+        // Avoid use same OpAddEntry between different ledger handle
+        int pendingSize = pendingAddEntries.size();
```

Comment:
hmm, this doesn't seem to be correct to me. you need to preserve the order when adding the newly created ops back to the queue.

what you need to do:

- drain the pendingAddEntries queue; 
- create a new OpAddEntry for each entry
- add these ops into an intermediate list in the order of how they are drained
- after the pendingAddEntries are drained, add the intermediate list back to the pendingAddEntries queue.

Reply:
The pendingAddEntries is ConcurrentLinkedQueue, there is no drainTo method in ConcurrentLinkedQueue, maybe we can use pendingAddEntries.toArray() and then create a new OpAddEntry for each item of the array and add the new entry to an intermediate list.

Reply:
I see. it is a queue here.